### PR TITLE
add new components selectLanguage and Usercard_Language

### LINKED ISSUE #557: 🔧 [Refactor] Switch language only in user profile
## 🔧 Refactor ticket

Switch language only in user profile not on every page.
-> Make sure the behaviour is correct when a new user registers
-> Is the browser language read correctly?

### PR DESCRIPTION
**Pull request by ogerly**
## 🍰 Pullrequest
an additional component was created to select the language. in the registration and in the profile area of the user's settings this component is used. 

### Issues
- fixes #557

### CODE REVIEW

File: frontend/src/views/Pages/Register.vue [hunk 1]
```diff
@@ -86,13 +86,14 @@
...
-                      {{ $t('language') }}
-                      <b-form-select
-                        id="selectedLanguage"
-                        v-model="selected"
-                        :options="options"
-                        class="mb-3"
+                      {{ $t('language') }} {{ language }}
+                      <!--   <b-form-select
+                        v-model="language"
+                        class="selectedLanguage mb-3"
```
Comment (ulfgebhardt) [inline frontend/src/views/Pages/Register.vue:92]: Can you remove the comment if this was the old implementation? Seems logical to me.

File: frontend/src/views/Pages/UserProfile/UserCard_Language.vue [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    class="bg-transparent pt-3 pb-3"
+    style="background-color: #ebebeba3 !important"
+  >
+    <b-container>
+      <div>
+        <b-row class="mb-4 text-right">
+          <b-col class="text-right">
+            <a @click="!editLanguage ? (editLanguage = !editLanguage) : cancelEdit()">
+              <span class="pointer mr-3">{{ $t('form.changeLanguage') }}</span>
+              <b-icon v-if="!editPassword" class="pointer ml-3" icon="pencil" />
```
Comment (ulfgebhardt) [inline frontend/src/views/Pages/UserProfile/UserCard_Language.vue:13]:
```suggestion
              <b-icon v-if="!editLanguage" class="pointer ml-3" icon="pencil" />
```
Is this correct?

**Review by ulfgebhardt — CHANGES_REQUESTED** [review]

File: frontend/src/views/Pages/UserProfile/UserCard_Language.vue (hunk 2 above)
Comment (ogerly) [reply frontend/src/views/Pages/UserProfile/UserCard_Language.vue]: no  the name is in all  show...[Password, Language, ...]

File: frontend/src/components/LanguageSwitchSelect.vue [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        { value: 'de', text: this.$t('languages.de') },
+        { value: 'en', text: this.$t('languages.en') },
+      ],
+    }
+  },
+  props: {
+    language: { type: String },
+  },
+  created() {
+    if (!this.register) this.selected = this.$store.state.language
```
Comment (Mogge) [inline frontend/src/components/LanguageSwitchSelect.vue:27]: Wo kommt das register her? Ist das nicht immer undefined?

File: frontend/src/views/Pages/UserProfile/UserCard_FormUsername.vue [hunk 4]
```diff
@@ -1,35 +1,27 @@
 <template>
   <b-card id="formusername" class="bg-transparent" style="background-color: #ebebeba3 !important">
-    <b-container v-if="username === ''">
```
Comment (Mogge) [inline frontend/src/views/Pages/UserProfile/UserCard_FormUsername.vue:3]: Why did you change this file?

File: frontend/src/views/Pages/UserProfile/UserCard_FormUserPasswort.vue [hunk 5]
```diff
@@ -1,48 +1,43 @@
...
-                  </b-button>
-                </div>
-              </b-col>
-            </b-row>
-          </b-form>
-        </validation-observer>
-      </div>
-    </b-container>
-  </div>
+  <b-card id="change_pwd" class="bg-transparent" style="background-color: #ebebeba3 !important">
```
Comment (Mogge) [inline frontend/src/views/Pages/UserProfile/UserCard_FormUserPasswort.vue:2]: Why did you change this file?

File: frontend/src/views/Pages/UserProfile/UserCard_FormUserData.vue [hunk 6]
```diff
@@ -1,81 +1,71 @@
 <template>
-  <div class="userdata_form" fluid="sm">
```
Comment (Mogge) [inline frontend/src/views/Pages/UserProfile/UserCard_FormUserData.vue:2]: Why did you change this file?

**Review by Mogge — CHANGES_REQUESTED** [review]
Ich verstehe nicht warum du so viele Dateien geändert hast. SInd das wirklich Verbesserungen? Ist das abgesprochen? Ich habe die Tests fürs Register gefixt.

**Review by Mogge — APPROVED** [review]
